python/ray/autoscaler/autoscaler.py

```diff
@@ -70,6 +70,8 @@ def __init__(self,
         else:
             self.instance_types = None
             self.resource_demand_scheduler = None
+        # node_id -> logical instance name mapping
+        self.node_types = {}
```

Comment:
I think this is redundant, since you can grab the instance "name" via node_tags(node_id)[TAG_RAY_INSTANCE_TYPE]. Btw, do you want to rename INSTANCE_TYPE to INSTANCE_NAME?

Actually, how about we keep instance type as the "logical" name (e.g., Ray instance type), and then in the code that refers to the physical instance type (e.g., m4.4xl, call those `physical_instance_type`)?